src/Aspire.Hosting.Azure/Bicep/openai.bicep

```diff
@@ -0,0 +1,75 @@
+// c.f., https://learn.microsoft.com/azure/ai-services/create-account-bicep
+
+param name string
+param principalId string
+param principalType string = 'ServicePrincipal'
+param deployments array = []
+
+@description('Tags that will be applied to all resources')
+param tags object = {}
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+@allowed([ 'Enabled', 'Disabled' ])
+param publicNetworkAccess string = 'Enabled'
+
+@allowed([
+  'S0'
+])
+param sku string = 'S0'
+
+param allowedIpRules array = []
+param networkAcls object = empty(allowedIpRules) ? {
+  defaultAction: 'Allow'
+} : {
+  ipRules: allowedIpRules
+  defaultAction: 'Deny'
+}
+
+var resourceToken = uniqueString(resourceGroup().id)
+var accountName = '${name}-${resourceToken}'
+
+resource account 'Microsoft.CognitiveServices/accounts@2023-05-01' = {
+  name: accountName
+  location: location
+  sku: {
+    name: sku
+  }
+  kind: 'OpenAI'
+  properties: {
+    customSubDomainName: accountName
+    publicNetworkAccess: publicNetworkAccess
+    networkAcls: networkAcls
+  }
+  tags: tags
+}
+
+@batchSize(1)
```

Comment:
Why batchSize 1?

Reply:
This ensures that these are deployed sequentially as by default they are done in parallel. This was in the original template from @jongio so I assume this was recommended at least.

Reply:
You'll run into concurrency issues without it - this makes them serial.